react/src/Analyses/components/SelectPipelineStatus.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useMemo } from "react";
+import { MenuItem, Select } from "@material-ui/core";
+import { EditComponentProps } from "material-table";
+import { Analysis, PipelineStatus } from "../../typings";
+import { checkPipelineStatusChange } from "../../functions";
+
+export default function SelectPipelineStatus(props: EditComponentProps<Analysis>) {
+    const options = useMemo(
+        () =>
+            Object.values(PipelineStatus)
+                .filter(
+                    state => props.value === state || checkPipelineStatusChange(props.value, state)
+                )
+                .map(value => <MenuItem value={value}>{value}</MenuItem>),
+        [props.value]
+    );
```

Comment:
I'm noticing a side effect here that if you choose the wrong option, you have to back out of the entire edit to try again.